apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -0,0 +1,325 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+using Xunit;
+
+namespace Google.Cloud.Spanner.V1.Tests
+{
+    public class PgNumericTests
+    {
+        public static IEnumerable<object[]> GetRawValueAndScales()
+        {
+            yield return new object[] { BigInteger.Pow(10, 147455) - 1, 0, true };
+            yield return new object[] { BigInteger.Pow(10, 147455), 0, false };
+            yield return new object[] { -BigInteger.Pow(10, 147455), 0, false };
+            yield return new object[] { -BigInteger.Pow(10, 147455) + 1, 0, true };
+
+            yield return new object[] { BigInteger.Pow(10, 147455) - 1, 110, true };
+            yield return new object[] { BigInteger.Pow(10, 147455) + 1, 110, false };
+
+            yield return new object[] { ((BigInteger.Pow(10, 131072) - 1)/BigInteger.Pow(10, 16383)), 16383, true };
+            yield return new object[] { -((BigInteger.Pow(10, 131072) - 1)/BigInteger.Pow(10, 16383)), 16383, true };
+
+            yield return new object[] { ((BigInteger.Pow(10, 131072) - 1) / BigInteger.Pow(10, 16383)), 16384, false };
+            yield return new object[] { -((BigInteger.Pow(10, 131072) - 1) / BigInteger.Pow(10, 16383)), 16384, false };
+
+            yield return new object[] { BigInteger.Pow(10, 147455) - 1, 16383, true };
+            yield return new object[] { -BigInteger.Pow(10, 147455) + 1, 16383, true };
+        }
+
+        public static IEnumerable<object[]> GetStringInputs()
+        {
+            yield return new object[] { PgNumeric.MaxValue.ToString(), true };
+            yield return new object[] { PgNumeric.MinValue.ToString(), true };
+            yield return new object[] { (BigInteger.Pow(10, 147455) - 1).ToString(), true };
+            yield return new object[] { (-(BigInteger.Pow(10, 147455) - 1)).ToString(), true };
+            yield return new object[] { (BigInteger.Pow(10, 147455)).ToString(), false };
+            yield return new object[] { (-(BigInteger.Pow(10, 147455) + 1)).ToString(), false };
+            yield return new object[] { PgNumeric.NaN.ToString(), true };
+        }
+
+        [Theory]
+        [InlineData("NaN", "NaN")]
+        [InlineData("Nan", "nan")]
+        [InlineData("0.00", "000")]
+        [InlineData("-1.0", "-1.000")]
+        [InlineData("-1", "-1.0")]
+        [InlineData("1", "1.0")]
+        [InlineData("0.00000009", "0.00000009")]
+        [InlineData("-0.00", "0.0")]
+        public void CompareToEqualValueTest(string input1, string input2)
+        {
+            var number1 = PgNumeric.Parse(input1);
+            var number2 = PgNumeric.Parse(input2);
+            var result = number1.CompareTo(number2);
+            Assert.Equal(0, result);
+        }
+
+        [Theory]
+        [InlineData("9999999999999", "NaN")]
```

Comment:
Nan should be compared aginst maxValue and minValue, and zero, and a negative value as well.